src/main/java/org/synyx/urlaubsverwaltung/application/specialleave/SpecialLeaveSettingsService.java

```diff
@@ -0,0 +1,50 @@
+package org.synyx.urlaubsverwaltung.application.specialleave;
+
+import org.springframework.data.util.Streamable;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Map;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+
+@Component
+public class SpecialLeaveSettingsService {
+
+    private final SpecialLeaveSettingsRepository specialLeaveSettingsRepository;
+
+    SpecialLeaveSettingsService(SpecialLeaveSettingsRepository specialLeaveSettingsRepository) {
+        this.specialLeaveSettingsRepository = specialLeaveSettingsRepository;
+    }
+
+    public void saveAll(List<SpecialLeaveSettingsItem> specialLeaveSettingsItems) {
+
+        final Map<Integer, SpecialLeaveSettingsItem> itemsById = Streamable.of(specialLeaveSettingsItems).stream()
+            .collect(toMap(SpecialLeaveSettingsItem::getId, entity -> entity));
+
+        final List<SpecialLeaveSettingsEntity> specialLeaveSettingsEntities = Streamable.of(specialLeaveSettingsRepository.findAll()).stream()
```

Comment:
`specialLeaveSettingsRepository.findAll()`

wir haben bereits die IDs. können also auch nur die holen die wir brauchen.

die aktuelle Implementierung kann auch kaputt gehen mit einer `NullPointer`. Wenn in `specialLeaveSettingsItems` eine fehlt, die aber in der DB ist. Dann ist `final SpecialLeaveSettingsItem specialLeaveSettingsItem = itemsById.get(entity.getId());` in Zeile 38 das item `null`

Reply:
do we need this or just save it?